instance/dig.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2021 The Dank Grinder authors.
+//
+// This source code has been released under the GNU Affero General Public
+// License v3.0. A copy of this license is available at
+// https://www.gnu.org/licenses/agpl-3.0.en.html
+//
+// Dig Functionality added by https://github.com/V4NSH4J
+
+package instance
+
+import (
+	"math/rand"
+	"regexp"
+
+	"github.com/dankgrinder/dankgrinder/discord"
+	"github.com/dankgrinder/dankgrinder/instance/scheduler"
+)
+
+func isSameLength(s1 string, s2 string) bool { // Function checks length of strings
+
+	if len(s1) == len(s2) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func hasSameLetters(s1 string, s2 string) bool { // Function checks the letters of the strings (Scrambled and Unscrambled Words)
+	var counter int
+	for _, letter1 := range s1 {
+		for _, letter2 := range s2 {
+			if letter1 == letter2 {
+				counter++
+				break
+			}
+		}
+	}
+	if counter == len(s1) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func find(tt string, stuff []string) (string, string) { //Some amazing stolen/borrowed code which finds the missing word and phrase from a list of pre-determined phrases
+	re := regexp.MustCompile("_")
+	q := re.ReplaceAllString(tt, `(\w+)`)
+	re2 := regexp.MustCompile(q)
+	for _, s := range stuff {
+		found := re2.FindStringSubmatch(s)
+		if len(found) > 0 {
+			return s, found[1]
+		}
+	}
+	return "", ""
+}
+
+func (in *Instance) digEventScramble(msg discord.Message) { // Dig Event Unscramble
+	scramble := exp.digEventScramble.FindStringSubmatch(msg.Content)[1] // Scrambled word is at index 1 of the search query
+
+	var Unscrambled []string // Unscrambled Solution
+
+	for _, word := range in.Compat.AllowedScrambles { // Allowed Scrambles in Compatibility
+		if isSameLength(scramble, word) && hasSameLetters(scramble, word) {
+			Unscrambled = append(Unscrambled, word)
```

Comment:
Why is `Unscrambled` a slice? Why is `Unscrambled` declared before it is defined? Why is `Unscrambled` even a variable?

Doing something like this would suffice. No need for the variable.
```go
for _, word := range in.Compat.AllowedScrambles {
		if len(scramble) == len(word) && haveSameChars(scramble, word) {
			in.sdlr.ResumeWithCommandOrPrioritySchedule(&scheduler.Command{
				Value: word,
				Log:   "responding to dig scramble",
			})
			return
		}
	}
```